core/pow/engine.go

```diff
@@ -251,20 +243,27 @@ func (e *Engine) DeliverTx(tx abci.Tx) error {
 	e.heightToTid[tx.BlockHeight()] = append(e.heightToTid[tx.BlockHeight()], tx.GetPoWTID())
 	e.seenTid[tx.GetPoWTID()] = struct{}{}
 
-	// if we've not seen any transactions from this party this block then let it pass and save the observed difficulty
-	if _, ok := state.blockPartyToSeenCount[tx.Party()]; !ok {
-		state.blockPartyToObservedDifficulty[tx.Party()] = uint(d)
-		state.blockPartyToSeenCount[tx.Party()] = 1
+	// if it's the first transaction we're seeing from any party for this block height, initialise the state
+	if _, ok := state.blockToPartyState[txBlock]; !ok {
```

Comment:
Is `txBlock` here the height of the block being created? As we should be using the height the PoW was generated using, but looks like this might be the new height?

Reply:
its the height of the block pow was generated with not the height of the block it is included in. txBlock = tx.BlockHeight

Reply:
Ok cool, wasn't sure whether `tx` here was the one pow used or the new one being created 👍